docs/src/content/docs/it/guides/site-search.mdx

```diff
@@ -0,0 +1,165 @@
+
+Per escludere una pagina dall'indice di ricerca, aggiungi [`pagefind: false`](/it/reference/frontmatter/#pagefind) al frontmatter della pagina:
+
+## Nascondere contenuti dai risultati di ricerca
+
+### Escludere una pagina
+
+Per escludere una pagina dall'indice di ricerca, aggiungi [`pagefind: false`](/reference/frontmatter/#pagefind) nel frontmatter della pagina:
+
+```md title="src/content/docs/not-indexed.md" ins={3}
+---
+title: Contenuto da nascondere dalla ricerca
+pagefind: false
+---
+```
+
+### Escludere parte di una pagina
+
+Pagefind ignorerà il contenuto all'interno di un elemento con l'attributo [`data-pagefind-ignore`](https://pagefind.app/docs/indexing/#removing-individual-elements-from-the-index).
+
+Nell'esempio seguente, il primo paragrafo verrà mostrato nei risultati di ricerca, ma il contenuto del `<div>` no:
+
+```md title="src/content/docs/partially-indexed.md" ins="data-pagefind-ignore"
+---
+title: Pagina indicizzata parzialmente
+---
+
+Questo testo sarà ricercabile.
+
+<div data-pagefind-ignore>
+
+  Questo testo sarà nascosto dalla ricerca. 
+
+</div>
+```
+
+## Provider di ricerca alternativi
+
+### Algolia DocSearch
+
+Se hai accesso al programma [Algolia's DocSearch](https://docsearch.algolia.com/) e vuoi usarlo al posto di Pagefind, puoi utilizzare il plugin ufficiale Starlight DocSearch.
+
+1. Installa `@astrojs/starlight-docsearch`:
+
+    <Tabs>
+
+    <TabItem label="npm">
+
+    ```sh
+    npm install @astrojs/starlight-docsearch
+    ```
+
+    </TabItem>
+
+    <TabItem label="pnpm">
+
+    ```sh 
+    pnpm add @astrojs/starlight-docsearch
+    ```
+
+    </TabItem>
+
+    <TabItem label="Yarn">
+
+    ```sh
+    yarn add @astrojs/starlight-docsearch
+    ```
+
+    </TabItem>
+
+    </Tabs>
+
+2. Aggiungi DocSearch alla configurazione [`plugins`](/it/reference/configuration/#plugins) di Starlight in `astro.config.mjs` e passagli il tuo `appId`, `apiKey` e `indexName` di Algolia:
+
+   ```js ins={4,10-16} 
+   // astro.config.mjs
+   import { defineConfig } from 'astro/config';
+   import starlight from '@astrojs/starlight';
+   import starlightDocSearch from '@astrojs/starlight-docsearch';
+   
+   export default defineConfig({
+     integrations: [
+       starlight({ 
+         title: 'Sito con DocSearch',
+         plugins: [
+           starlightDocSearch({
+             appId: 'TUO_APP_ID',
+             apiKey: 'TUA_API_KEY_DI_RICERCA', 
+             indexName: 'TUO_NOME_INDICE'
+           }),
+         ],  
+       }),
+     ],
+   });
+   ```
+   
+Con questa configurazione aggiornata, la barra di ricerca nel tuo sito aprirà ora una schermata Algolia invece della schermata di ricerca predefinita.
+
+#### Traduzione dell'interfaccia utente di DocSearch
+
+DocSearch fornisce solo stringhe dell'interfaccia utente in inglese di default.
+Aggiungi traduzioni della UI della schermata per la tua lingua utilizzando il sistema di internazionalizzazione integrato in Starlight.
```

Comment:
This sentence is missing a link to `/it/guides/i18n/#tradurre-linterfaccia-starlight`, I think it should be around `sistema di internazionalizzazione` but I prefer to let you add it at the proper place.